ulauncher/api/shared/action/LaunchAppAction.py

```diff
@@ -26,13 +27,22 @@ def keep_app_open(self):
     def run(self):
         app = read_desktop_file(self.filename)
         command = app.get_string('Exec')
+        
         terminal_exec = settings.get_property('terminal-command')
         if app.get_boolean('Terminal') and terminal_exec and command:
             logger.info('Run command %s (%s) in preferred terminal (%s)', command, self.filename, terminal_exec)
             subprocess.Popen(terminal_exec.split() + [pipes.quote(command)])
         else:
             logger.info('Run application %s (%s)', app.get_name(), self.filename)
             try:
-                app.launch()
+                if (command.startswith("pkexec")):
+                    subprocess.Popen(
+                        args=shlex.split('sh -c "'+command+'"'),
```

Comment:
`args=shlex.split('sh -c "'+command+'"'),` could be changed to `args=['sh', '-c', command],` couldn't it? Less code and you wouldn't need to import shlex.

Reply:
I imagine if you're suggesting so, that's probably true! I'm inexperienced with Python and tried to hack code together until it worked. My hope was that someone might see this, tidy it up/fix it a little and resolve the initial problem. ULauncher could completely replace the mint menu if this ever gets fixed.

Reply:
I haven't even tried the PR, but it should do exactly the same thing with these line swapped. I'm not using apps that need root permissions a lot.